hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/model/HoodieFlinkRecordMerger.java

```diff
@@ -21,11 +21,12 @@
 
 import org.apache.hudi.common.model.HoodieRecord;
 import org.apache.hudi.common.model.HoodieRecordMerger;
+import org.apache.hudi.common.model.OperationModeAwareness;
 
 /**
  * Base class for {@link HoodieRecordMerger } of HoodieFlinkRecord.
  */
-public abstract class HoodieFlinkRecordMerger implements HoodieRecordMerger {
+public abstract class HoodieFlinkRecordMerger implements HoodieRecordMerger, OperationModeAwareness {
```

Comment:
Remove this `OperationModeAwareness`, it is only designed for backward compatibility.

Reply:
This is for in-memory records merging during data ingestion (mainly for `DELETE` merging result)

If we modify the representation of merging result for DELETE in this PR, some refactoring of callers are needed, as currently the merging result is considered as DELETE if it's Option.empty, e.g., HoodieMergeHandle, FileGroupRecordBuffer etc. 

And other engine specific mergers should also be aligned on DELETE merging result if callers are refactored. Maybe we can create a separate ticket for the refactoring to keep the scope of this PR clean?